aptos-move/mvhashmap/src/versioned_aggregators.rs

```diff
@@ -423,50 +425,197 @@ impl VersionedAggregators {
     /// Upon commit Snapshot and Delta entries are all required to be replaced with Values.
     pub fn record_value(
         &self,
-        id: AggregatorID,
+        id: K,
         txn_idx: TxnIndex,
         value: AggregatorValue,
-        maybe_apply: Option<AggregatorApplyChange>,
+        maybe_apply: Option<AggregatorApplyChange<K>>,
     ) {
         self.values
             .get_mut(&id)
             .expect("VersionedValue for an (resolved) ID must already exist")
             .insert(txn_idx, AggregatorEntry::Value(value, maybe_apply));
     }
 
-    pub fn record_apply(&self, id: AggregatorID, txn_idx: TxnIndex, apply: AggregatorApplyChange) {
+    pub fn record_apply(&self, id: K, txn_idx: TxnIndex, apply: AggregatorApplyChange<K>) {
         self.values
             .get_mut(&id)
             .expect("VersionedValue for an (resolved) ID must already exist")
             .insert(txn_idx, AggregatorEntry::Apply(apply));
     }
 
-    pub fn mark_estimate(&self, id: AggregatorID, txn_idx: TxnIndex) {
+    pub fn mark_estimate(&self, id: K, txn_idx: TxnIndex) {
         self.values
             .get_mut(&id)
             .expect("VersionedValue for an (resolved) ID must already exist")
             .mark_estimate(txn_idx);
     }
 
-    pub fn remove(&self, id: AggregatorID, txn_idx: TxnIndex) {
+    pub fn remove(&self, id: K, txn_idx: TxnIndex) {
         self.values
             .get_mut(&id)
             .expect("VersionedValue for an (resolved) ID must already exist")
             .remove(txn_idx);
     }
 
-    pub fn update_committed_idx(&self, committed_idx: TxnIndex) {
+    pub fn try_commit(&self, idx_to_commit: TxnIndex, ids: Vec<K>) -> Result<(), CommitError> {
+        // we may not need to return values here, we can just read them.
+        use AggregatorApplyChange::*;
+
+        if idx_to_commit != self.next_idx_to_commit.load(Ordering::SeqCst) {
```

Comment:
I think it's reasonable to require the caller to call try_commit for all indices, even the ones that wouldn't touch any V2 aggregators. And it is visible from the code clearly, but maybe we should still document one liner before the method (I often look there for these preconditions)?